lib/ansible/modules/clustering/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: kubevirt_cdi_upload
+
+short_description: Upload local VM images to CDI Upload Proxy.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+
+description:
+  - Use Openshift Python SDK to create UploadTokenRequest objects.
+  - Transfer contents of local files to the CDI Upload Proxy.
+
+extends_documentation_fragment:
+  - k8s_auth_options
+
+options:
+  pvc_name:
+    description:
+      - Use to specify the name of the target PersistentVolumeClaim.
+    required: true
+  pvc_namespace:
+    description:
+      - Use to specify the namespace of the target PersistentVolumeClaim.
+    required: true
+  upload_host:
+    description:
+      - URL containing the host and port on which the CDI Upload Proxy is available.
+  upload_host_verify_ssl:
+    description:
+      - Whether or not to verify the CDI Upload Proxy's SSL certificates.
+    default: true
+    type: bool
+  path:
+    description:
+      - Path of local image file to transfer.
+  kind:
+    description:
+      - Required for compatibility with ansible <2.8. Can't be modified.
+    default: UploadTokenRequest
+    choices:
+      - UploadTokenRequest
+  api_version:
+    description:
+      - Required for compatibility with ansible <2.8. Can't be modified.
+    default: upload.cdi.kubevirt.io/v1alpha1
+    choices:
+      - upload.cdi.kubevirt.io/v1alpha1
+  src:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
```

Comment:
Why it's required for that compatibility?

Reply:
The base Kubernetes classes were created with only `k8s` module in mind. This has been corrected to a large degree with 2.8 (partially with my `k8s_service` patches), so it's much better on the devel branch of ansible, but in 2.7 it's very noticeable. If you don't define all those parameters, the module will just fail with 'key XYZ not found'.

Reply:
If this gets somehow submitted upstream, then yes, we'll need to clean it up. But that will be a whole different process, nobody's gonna let us not remove those args. :)